compiler/rustc_mir_build/src/build/scope.rs

```diff
@@ -1118,21 +1118,38 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
     }
 
     /// Utility function for *non*-scope code to build their own drops
+    /// Force a drop at this point in the MIR by creating a new block.
+    /// Should only be used for DropAndReplace.
+    /// Note that this does *not* insert the assignement in the MIR,
+    /// that should be added in the newly returned block by the caller.
     pub(crate) fn build_drop_and_replace(
         &mut self,
         block: BasicBlock,
+        statement_source_info: SourceInfo,
         span: Span,
         place: Place<'tcx>,
-        value: Operand<'tcx>,
+        value: Rvalue<'tcx>,
     ) -> BlockAnd<()> {
         let source_info = self.source_info(span);
         let next_target = self.cfg.start_new_block();
 
+        let assign = self.cfg.start_new_cleanup_block();
+        self.cfg.push_assign(assign, statement_source_info, place, value.clone());
+        // We still have to build the scope drops so we don't know which block will follow.
+        // This terminator will be overwritten once the unwind drop tree builder runs.
+        self.cfg.terminate(assign, statement_source_info, TerminatorKind::Unreachable);
```

Comment:
Is this necessary? Since the terminator will be added later, it can be left "uninitialized" (it is internally an `Option::None`)